### PR TITLE
[FIX] 75674 Website not showing prices correctly when using discounts

### PR DESCRIPTION
**Pull request by yaazkal**
All information related to the issue is at https://github.com/odoo/odoo/issues/75674 

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website_sale/models/product.py [hunk 1]
```diff
@@ -301,7 +301,10 @@ def _get_combination_info(self, combination=False, product_id=False, add_qty=1,
             price = taxes.compute_all(combination_info['price'], pricelist.currency_id, quantity_1, product, partner)[tax_display]
             if pricelist.discount_policy == 'without_discount':
                 combination_info['list_price'] = self.env['account.tax']._fix_tax_included_price_company(combination_info['list_price'], product.sudo().taxes_id, taxes, company_id)
-                list_price = taxes.compute_all(combination_info['list_price'], pricelist.currency_id, quantity_1, product, partner)[tax_display]
+                if pricelist.item_ids.base == 'pricelist' and pricelist.item_ids.base_pricelist_id:
```
Comment (Williambraecky) [inline addons/website_sale/models/product.py:304]: This will fail with multiple pricelist_items.

File: addons/website_sale/models/product.py [hunk 2]
```diff
@@ -301,7 +301,10 @@ def _get_combination_info(self, combination=False, product_id=False, add_qty=1,
...
+                    list_price = pricelist.item_ids.base_pricelist_id._compute_price_rule([(product, quantity_1, partner)])[product.id][0]  # TDE: 0 = price, 1 = rule
```
Comment (Williambraecky) [inline addons/website_sale/models/product.py:305]: What about the taxes?

**Review by Williambraecky — CHANGES_REQUESTED** [review]
Hello thanks for the fix!
Unfortunately it isn't acceptable as is.

I am not that knowledgeable about pricelist but I have the feeling this should be fixed in the base `_get_combination_info` instead.

@Feyensv could you take a look on this one if you have some time?
Cheers

**Comment by Feyensv** [comment]
@Williambraecky Well, it's a tricky problem to fix in stable (even more if we want to do it cleanly).
The discount computation (when considering recursive pricelists) is not coherent between the different scopes at all (cf #91010 for the same problem in POS).

I'm not even sure it's fully fixed in master (the SO prices will be correct, but the displayed discount might not be).
I'm thinking of redoing the whole `_get_combination_info` logic, but I wasn't convinced by my first try ^^.

Ideally, the logic of `_get_real_price_currency` should be extracted in a separate util method and used everywhere to consistently compute the pricelist discounts.

**Comment by C3POdoo** [comment]
Dear @yaazkal,

Thank you for your contribution but the version 13.0 is no longer supported.
We only support the last 3 stable versions so no longer accepts patches into this branch.

We apology if we could not look at your request in time.
If the contribution still makes sense for the upper version, please let us know and do not hesitate to recreate one for the recent versions. We will try to check it as soon as possible.

_This is an automated message._

**Pull request closed without merging**